upstairs/src/downstairs.rs

```diff
@@ -3166,6 +3174,30 @@ impl Downstairs {
             self.ackable_work.insert(ds_id);
         }
 
+        if job.reply_time.iter().all(Option::is_some) && !job.replay {
+            let fastest_id = ClientId::iter()
+                .min_by_key(|i| job.reply_time[*i].unwrap())
+                .unwrap();
+            let slowest_time = *job.reply_time.iter().flatten().max().unwrap();
+
+            // Apply a delay to the fastest client, and clear the delay for the
+            // other two clients.
+            for i in ClientId::iter() {
+                let delay_time_us: u64 = if i == fastest_id {
+                    // 0 delay below 10ms of lead time, then linearly increasing
+                    // to a maximum of 10 ms.  These are all roughly-eyeballed
+                    // numbers!
+                    let dt = slowest_time - job.reply_time[i].unwrap();
+                    let lead_time_us: u64 = dt.as_micros().try_into().unwrap();
+                    (lead_time_us.saturating_sub(10_000) / 100).min(10_000)
+                } else {
+                    0
+                };
+
+                self.clients[i].set_delay_us(delay_time_us);
+            }
```

Comment:
Is there any way we can see this from the outside?  It would be interesting to be able
to track this and watch how it changes, but I'm not sure how to bubble it out to
where we could grab it with the rest of the `collect_stats()`.

Reply:
I added this to the existing `up-status` DTrace probe in 597b9500b147b344dfdcb1d65190bc505a12c11b

(The string is now 768 bytes long with all zeros, so still well below the `strsize=1k` limit)